modules/dhp_connected_devices/spec/controllers/dhp_connected_devices/fitbit/fitbit_controller_spec.rb

```diff
@@ -75,34 +75,80 @@ def fitbit_callback(params = '')
-        faraday_response = double('response', status: 200, body: '{ "access_token": "some token" }')
-        allow_any_instance_of(Faraday::Connection).to receive(:post).with(anything).and_return(faraday_response)
-        allow(VeteranDeviceRecordsService).to receive(:create_or_activate).with(anything, anything).and_raise(
-          ActiveRecord::ActiveRecordError
-        )
+      it "redirects with 'fitbit=error' when authorization is given but token exchange is unsuccessful" do
+        allow_any_instance_of(fitbit_client).to receive(:get_auth_code).with(any_args).and_return('889709')
+        allow_any_instance_of(fitbit_client).to receive(:get_token).with(any_args).and_raise(token_exchange_error)
+        allow_any_instance_of(SentryLogging).to receive(:log_exception_to_sentry).with(any_args)
 
-        expect_any_instance_of(SentryLogging).to receive(:log_exception_to_sentry).with(
-          instance_of(ActiveRecord::ActiveRecordError),
-          { icn: current_user.icn }
-        )
-        expect(fitbit_callback('?code=889709')).to redirect_to 'http://localhost:3001/health-care/connected-devices/?fitbit=error#_=_'
+        expect(token_storage_service).not_to receive(:store_tokens)
+
+        expect(fitbit_callback('?code=889709')).to redirect_to error_path
+      end
+
+      it "redirects with 'fitbit=error' when token exchange is successful but token storage is unsuccessful" do
+        allow_any_instance_of(fitbit_client).to receive(:get_auth_code).with(any_args).and_return('889709')
+        allow_any_instance_of(fitbit_client).to receive(:get_token).with(any_args).and_return(access_token)
+        allow_any_instance_of(TokenStorageService).to receive(:store_tokens).with(any_args).and_return(false)
+
+        expect(fitbit_callback('?code=889709')).to redirect_to error_path
+      end
+
+      it "redirects with 'fitbit=error' when token exchange is successful but error is thrown during token storage" do
+        allow_any_instance_of(fitbit_client).to receive(:get_auth_code).with(any_args).and_return('889709')
+        allow_any_instance_of(fitbit_client).to receive(:get_token).with(any_args).and_return(access_token)
+        allow_any_instance_of(TokenStorageService).to receive(:store_tokens).with(any_args).and_raise(TokenStorageError)
+        allow_any_instance_of(SentryLogging).to receive(:log_exception_to_sentry).with(any_args)
+
+        expect(fitbit_callback('?code=889709')).to redirect_to error_path
+      end
+
+      it "redirects with 'fitbit=success' when is token storage is successful'" do
+        allow_any_instance_of(fitbit_client).to receive(:get_auth_code).with(any_args).and_return('889709')
+        allow_any_instance_of(fitbit_client).to receive(:get_token).with(any_args).and_return(access_token)
+        allow_any_instance_of(TokenStorageService).to receive(:store_tokens).with(any_args).and_return(true)
+        allow_any_instance_of(SentryLogging).to receive(:log_exception_to_sentry).with(any_args)
```

Comment:
Nitpick comment: this SentryLogging mocking is not needed in this case.

Reply:
Removed